### ISSUE TITLE
Cannot login with Safari

### ISSUE THREAD

**Issue by stonerl** [owncloud/core#15780]
### Steps to reproduce
1. Open Safari
2. Login to your ownCloud.

### Expected behaviour
The user should be logged in and the file-app should be shown.

### Actual behaviour
Safari shows an error-message: "Safari Can't Open the Page ... because the server unexpectedly dropped the connection." After 10 attempts I was able to log in. Firefox and Chrome don't show this error.

### Server configuration
**Operating system**:
Ubuntu 14.04.2 LTS

**Web server:**
Apache/2.4.7 (Ubuntu)

**Database:**
mysql  Ver 14.14 Distrib 5.5.41, for debian-linux-gnu (x86_64) using readline 6.3

**PHP version:**
PHP 5.5.9-1ubuntu4.9

**ownCloud version:** (see ownCloud admin page)
8.0.2

**Updated from an older ownCloud or fresh install:**
updated from 7.0.4 but error appeared also with this Version

**List of activated apps:**
- Activity
- Calendar
- Contacts
- Deleted Files
- Documents
- External User Support
- File Locking
- First Run Wizard
- Full Text Search
- PDF Viewer
- Provisioning
- Share Files
- Text Editor
- Versions

**The content of config/config.php:**

```<?php
$CONFIG = array (
  'instanceid' => 'oc91218af27b',
  'passwordsalt' => '###',
  'trusted_domains' => 
  array (
    0 => '###',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'dbtype' => 'mysql',
  'version' => '8.0.2.0',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'dbpassword' => '###',
  'installed' => true,
  'mail_from_address' => 'owncloud',
  'mail_smtpmode' => 'sendmail',
  'loglevel' => 0,
  'logtimezone' => 'Europe/Berlin',
  'log_type' => 'owncloud',
  'log_authfailip' => true,
  'mail_domain' => '###',
  'default_language' => 'de_DE',
  'user_backends' => 
  array (
    0 => 
    array (
      'class' => 'OC_User_IMAP',
      'arguments' => 
      array (
        0 => '{###:993/imap/ssl/novalidate-cert}',
      ),
    ),
  ),
  'maintenance' => false,
  'overwritewebroot' => '/owncloud',
  'forcessl' => true,
  'appcodechecker' => false,
  'session_keepalive' => true,
  'remember_login_cookie_lifetime' => 2592000,
  'secret' => '###',
  'theme' => '',
  'skeletondirectory' => '',
  'overwrite.cli.url' => '###',
  'asset-pipeline.enabled' => true,
  'allowZipDownload' => true,
  'xframe_restriction' => true,
);
```

**Are you using external storage, if yes which one:**
No

**Are you using encryption:**
No

**Are you using an external user-backend, if yes which one:**
OC_User_IMAP

### Client configuration
**Browser:** 
Safari 8.0.5

**Operating system:** 
OS X 10.10.3

#### ownCloud log (data/owncloud.log)
```
 Exception: {"Message":"No basic authentication headers were found","Code":0,"Trace":"#0 \/var\/www\/owncloud\/lib\/private\/connector\/sabre\/auth.php(126): Sabre\\DAV\\Auth\\Backend\\AbstractBasic->authenticate(Object(OC_Connector_Sabre_Server), 'ownCloud')\n#1 \/var\/www\/owncloud\/lib\/private\/connector\/sabre\/auth.php(106): OC_Connector_Sabre_Auth->auth(Object(OC_Connector_Sabre_Server), 'ownCloud')\n#2 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/Sabre\/DAV\/Auth\/Plugin.php(108): OC_Connector_Sabre_Auth->authenticate(Object(OC_Connector_Sabre_Server), 'ownCloud')\n#3 [internal function]: Sabre\\DAV\\Auth\\Plugin->beforeMethod('PROPFIND', '')\n#4 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/Sabre\/DAV\/Server.php(433): call_user_func_array(Array, Array)\n#5 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/Sabre\/DAV\/Server.php(455): Sabre\\DAV\\Server->broadcastEvent('beforeMethod', Array)\n#6 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/Sabre\/DAV\/Server.php(214): Sabre\\DAV\\Server->invokeMethod('PROPFIND', '')\n#7 \/var\/www\/owncloud\/apps\/files\/appinfo\/remote.php(61): Sabre\\DAV\\Server->exec()\n#8 \/var\/www\/owncloud\/remote.php(54): require_once('\/var\/www\/ownclo...')\n#9 {main}","File":"\/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/Sabre\/DAV\/Auth\/Backend\/AbstractBasic.php","Line":73} 
```

**Comment by karlitschek** [comment]
Not reproducible here. This is most likely an error with you setup. Maybe a hickup in the imap server?

**Comment by stonerl** [comment]
Hmm. The strange thing is that this "error" only happens when I use Safari. Firefox, Opera, Chrome work just fine. And everytime the connection gets dropped the above ownCloud.log entry appears.

**Comment by tflidd** [comment]
Can you reach other SSL-sites on your server? Or, can you access owncloud without SSL?
There is also a error console for safari:
* Open Safari and choose Safari > Preferences > Advanced. Then select Show Develop menu in menu bar.
* Choose Develop > Show Error Console.
* Close the console by clicking the close button on the left hand side of the error consoles menu bar.

**Comment by stonerl** [comment]
@tflidd 
I can reach other SSL-Sites on my server. Since I enforce SSL neither ownCloud nor any other site can be accessed without SSL. If I disable this then of course ownCloud would be accessible without SSL.

The only error I get from the Console, when the connection gets dropped is:
```
Failed to load resource: The network connection was lost.
``` 

The problem is that I cannot reliably reproduce this error. Sometimes I can login 10 times without any problem.

My first impression was that this might only be an error that appears with IMAP-Users, but this isn't the case. Also when a local user, e.g. the admin, tries to login this error happens. But I wonder if this might be a problem with Safari itself. As I said before; I never ever encountered this behavior with any other Browser.

**Comment by stonerl** [comment]
This might be a Safari problem:

http://mandarapte.com/apple/fix-error-safari-cant-open-page-server-unexpectedly-dropped-connection-occurs-server-busy/

https://www.igorkromin.net/index.php/2014/11/27/how-to-fix-safari-cant-open-the-page-error/

But clearing the cookies and cache did not solve this for my it still happens randomly

**Comment by stonerl** [comment]
Just another update. When I disable forcessl in the ownCloud config and connect via plain http this error never appears. As soon as I turn on SSL the hick-ups start again.

Another project had this issue as well:
https://github.com/Eugeny/ajenti/issues/378#issuecomment-43982529

**Comment by Sugaroverdose** [comment]
Try to disable sslv3 proto and ciphers in apache, safari still supports it but with some changes to fix poodle vulnerability, and it doesn't looks like it works properly

**Comment by stonerl** [comment]
@Sugaroverdose sorry for the late response. 

SSLv3 is already disabled and these are my cipher settings:
SSLCipherSuite HIGH:!aNULL:!MD5:!3DES:!AES128:!CAMELLIA:!kRSA

**Comment by Sugaroverdose** [comment]
Try to run https://www.ssllabs.com/ssltest/ it can help to troubleshoot common ssl issues.
Also you can try my cipher suite string, it's secure, but relies on DH same or bigger than 2048:
```
ECDH+AESGCM256:DH+AESGCM256:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES128:RSA+AESGCM256:RSA+AES256:RSA+AESGCM128:RSA+AES128:DH+3DES:RSA+3DES:!aNULL:!MD5:!DSS
```
Can't reproduce on OS X 10.10.3/Safari 8.0.5

**Comment by stonerl** [comment]
@Sugaroverdose 
Had done this already. But I'm pretty sure I found the problem.

I had to change the "KeepAliveTimeout" in the apache settings. The default value is 5. But for reasons I don't understand, setting the value between 1-9 will result in Safari dropping the connection. So I set it to 10 and Safari does not drop a single connection anymore.

Strange thing is, on my second server the value is still 5 and Safari never dropped the connection.
